### PR TITLE
REG-43: Review and improve automated test coverage

### PR DESCRIPTION
**Pull request by cpcundill**
Brought overall test coverage of project up to 55% from 12%.  This main area for improvement is within the `repository-hbase` module which is currently around the 25% mark.

### CODE REVIEW

**Review by ameenhaq — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -51,38 +50,86 @@ brew install sbt
...
+To run the `sbr-admin-api`, run the following:
 
 ``` shell
 sbt "api/run -Dsource=hbaseInMemory -Dsbr.hbase.inmemory=true"
 ```
 
-Note: When using In-Memory HBase, make sure the path to the project path does not include spaces.
+| Environment Variable | Default Value   | Valid Values                                              |
+|----------------------|-----------------|-----------------------------------------------------------|
+| validation.id        | ".{3,8}"        | Any regex string for validating the id                    |
```
Comment (ONS-Tom) [inline README.md:61]: Some of these need updating

File: app/Module.java [hunk 2]
```diff
@@ -41,9 +41,9 @@ public Module(Environment environment, Configuration configuration) {
 
     @Override
     public void configure() {
-        if (configuration.getBoolean("database.in.memory")) {
+        if (configuration.getBoolean("hbase.in.memory")) {
```
Comment (ONS-Tom) [inline app/Module.java:44]: Should we convert this into Scala?

File: app/Module.java [hunk 3]
```diff
@@ -59,8 +59,8 @@ public void configure() {
 class RepositoryInitializer {
 
     @Inject
-    public RepositoryInitializer(Configuration configuration, AdminDataLoad dataLoader) throws Exception {
-        dataLoader.load(configuration.getString("database.table"), "201706", configuration.getString("csv.file"));
+    public RepositoryInitializer(Configuration configuration, AdminDataLoad dataLoader) {
+        dataLoader.load(configuration.getString("hbase.table.name"), "201706", configuration.getString("csv.file"));
```
Comment (ONS-Tom) [inline app/Module.java:63]: Are we supposed to hard code the period?

File: conf/routes [hunk 4]
```diff
@@ -3,18 +3,19 @@
...
 
 # Searching for a specific period
-#GET        /v1/periods/:period/records/:id              controllers.v1.AdminDataController.lookup(period: Option[String], id)
+GET        /v1/periods/:period/records/:id              controllers.v1.AdminDataController.lookup(period: Option[String], id)
 
-#GET        /v1/rest/periods/:period/records/:id         controllers.v1.AdminDataController.lookupRest(period: Option[String] ?= None, id)
-
-
-GET        /v1/records/:id/periods                 controllers.v1.AdminDataController.lookupRest(period: Option[String] ?= None, id)
+GET        /v1/records/:id/periods                      controllers.v1.AdminDataController.lookupRest(period: Option[String] ?= None, id)
```
Comment (ONS-Tom) [inline conf/routes:11]: Are we using this?

**Review by ONS-Tom — APPROVED** [review]

**Pull request merged**